php/libraries/User.class.inc

```diff
@@ -317,6 +317,24 @@ class User extends UserPermissions
     }
 
     /**
+     * Checks if the user is in at least one study site
+     *
+     * @return boolean
+     */
+    function hasStudySite()
+    {
+        $site_arr = $this->getCenterIDs();
+        foreach ($site_arr as $key=>$val) {
+            $site[$key]        = & Site::singleton($val);
+            $isStudySite[$key] = $site[$key]->isStudySite();
+            if (in_array("1", $isStudySite)) {
+                return true;
+            }
```

Comment:
can't this whole loop just be:

```
foreach ($site_arr as $key => $sitename) {
    $site = Site::singleton($sitename);
    if ($site->isStudySite()) {
        return true;
    }
}
```

I don't see the point of creating the arrays which aren't really used.

Reply:
me neither! total oversight (just simply copied and pasted the function that initially was in the codebase without cleaning it). Fixed now. Thanks.